### PR TITLE
Optimising the ToString operation

### PR DESCRIPTION
**Pull request by robertsipka**
Create new ecma-string from positive integers without cast it to ecma_number

JerryScript-DCO-1.0-Signed-off-by: Robert Sipka rsipka.uszeged@partner.samsung.com

### CODE REVIEW

File: jerry-core/ecma/operations/ecma-conversion.c [hunk 1]
```diff
@@ -355,7 +355,19 @@ ecma_op_to_string (ecma_value_t value) /**< ecma value */
...
+      {
+        ecma_number_t num = ecma_get_number_from_value (value);
+        res_p = ecma_new_ecma_string_from_number (num);
+      }
+      else
+      {
+        res_p = ecma_new_ecma_string_from_uint32 ((uint32_t) (ecma_get_integer_from_value (value)));
+      }
+    }
+    else if (ecma_is_value_float_number (value))
```
Comment (zherczeg) [inline jerry-core/ecma/operations/ecma-conversion.c]: I would call ecma_get_integer_from_value (value) first, then check the sign, and that makes ecma_get_number_from_value call unnecessary.

Comment (robertsipka) [inline jerry-core/ecma/operations/ecma-conversion.c]: Yes, indeed, thanks! I've updated this patch.

File: jerry-core/ecma/operations/ecma-conversion.c [hunk 2]
```diff
@@ -355,7 +355,19 @@ ecma_op_to_string (ecma_value_t value) /**< ecma value */
       res_p = ecma_get_string_from_value (value);
       res_p = ecma_copy_or_ref_ecma_string (res_p);
     }
-    else if (ecma_is_value_number (value))
+    else if (ecma_is_value_integer_number (value))
+    {
+      ecma_integer_value_t num = ecma_get_integer_from_value (value);
+      if (num < 0)
```
Comment (zherczeg) [inline jerry-core/ecma/operations/ecma-conversion.c]: I would add a newline before this. After that LGTM.

Comment (robertsipka) [inline jerry-core/ecma/operations/ecma-conversion.c]: Added, thanks.

File: jerry-core/ecma/operations/ecma-conversion.c [hunk 3]
```diff
@@ -352,10 +352,23 @@ ecma_op_to_string (ecma_value_t value) /**< ecma value */
 
     if (ecma_is_value_string (value))
     {
-      res_p = ecma_get_string_from_value (value);
+      res_p = ecma_get_string_from_value (value);ecma-array-object.c
```
Comment (LaszloLango) [inline jerry-core/ecma/operations/ecma-conversion.c]: accident change?

Comment (robertsipka) [inline jerry-core/ecma/operations/ecma-conversion.c]: Sadly, fixed.

**Comment by LaszloLango** [comment]
LGTM

**Comment by zherczeg** [comment]
LGTM

**Comment by robertsipka** [comment]
rebased with master

**Pull request merged**